canonical/charmed-kubeflow-uats
# Rewrite e2e test

Changes:
- Use mlflow 2.15.1
- Use kserver 0.13.1
- Use kfp v2.5.0
- Use new syntax from kfp v2
- Replace seldon with kserve

**Prerequisites:**
- Deploy charmed Kubeflwo https://charmed-kubeflow.io/docs/get-started-with-charmed-kubeflow
- Deploy charmed mlflow https://documentation.ubuntu.com/charmed-mlflow/en/latest/tutorial/mlflow/
- Deploy reesource dispatcher and relate everything needed https://documentation.ubuntu.com/charmed-mlflow/en/latest/tutorial/mlflow-kubeflow/

**UAT test (this can take very long):** 
- From the juju machine make sure yuou have kubectl set. 
- Download uats repo and switch to this branch. 
- Instal python3.8 (instructions in uats readme).
- run tox -e testenv:uats-local (this will run all uats also with e2e wine).

```
test_notebooks.py::test_notebook[e2e-wine-kfp-mlflow-kserve] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running e2e-wine-kfp-mlflow-kserve.ipynb...
PASSED                                                                   [ 11%]
test_notebooks.py::test_notebook[katib-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running katib-integration.ipynb...
PASSED                                                                   [ 22%]
test_notebooks.py::test_notebook[kfp-v1-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running kfp-v1-integration.ipynb...
PASSED                                                                   [ 33%]
test_notebooks.py::test_notebook[kfp-v2-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running kfp-v2-integration.ipynb...
PASSED                                                                   [ 44%]
test_notebooks.py::test_notebook[kserve-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running kserve-integration.ipynb...
PASSED                                                                   [ 55%]
test_notebooks.py::test_notebook[mlflow-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running mlflow-integration.ipynb...
PASSED                                                                   [ 66%]
test_notebooks.py::test_notebook[mlflow-kserve] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running mlflow-kserve.ipynb...
PASSED                                                                   [ 77%]
test_notebooks.py::test_notebook[mlflow-minio-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running mlflow-minio-integration.ipynb...
PASSED                                                                   [ 88%]
test_notebooks.py::test_notebook[training-integration] 
-------------------------------- live log call ---------------------------------
INFO     test_notebooks:test_notebooks.py:44 Running training-integration.ipynb...
PASSED                                                                   [100%]
```

**Manual test**
- Create jupyter notebook (make sure to add mlflow config from deopdown)
- Clone UAts in the notebook 
- go to `tests/notebooks/e2e-wine` 
- Run the notebook cell by cell

![image](https://github.com/user-attachments/assets/c612a676-b6ad-4fcb-89a6-a71427145fdf)
